harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/simpleUrl/Harvester.java

```diff
@@ -0,0 +1,280 @@
+ *
+ * This harvester has been tested with CKAN search API.
+ */
+class Harvester implements IHarvester<HarvestResult> {
+    public static final String LOGGER_NAME = "geonetwork.harvester.json";
+
+    private final AtomicBoolean cancelMonitor;
+    private Logger log;
+    private SimpleUrlParams params;
+    private ServiceContext context;
+
+    @Autowired
+    GeonetHttpRequestFactory requestFactory;
+
+    /**
+     * Contains a list of accumulated errors during the executing of this harvest.
+     */
+    private List<HarvestError> errors = new LinkedList<HarvestError>();
+
+    public Harvester(AtomicBoolean cancelMonitor, Logger log, ServiceContext context, SimpleUrlParams params) {
+        this.cancelMonitor = cancelMonitor;
+        this.log = log;
+        this.context = context;
+        this.params = params;
+    }
+
+    public HarvestResult harvest(Logger log) throws Exception {
+        this.log = log;
+        log.debug("Retrieving simple URL: " + params.getName());
+
+        requestFactory = context.getBean(GeonetHttpRequestFactory.class);
+
+        String jsonResponse = retrieveUrl(params.url, log);
+        if (cancelMonitor.get()) {
+            return new HarvestResult();
+        }
+        log.debug("Response is: " + jsonResponse);
+
+        // TODO: Add support for XML or JSON
+        int numberOfRecordsToHarvest = -1;
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode jsonObj = objectMapper.readTree(jsonResponse);
+
+        if (StringUtils.isNotEmpty(params.numberOfRecordPath)) {
+            try {
+                numberOfRecordsToHarvest = jsonObj.at(params.numberOfRecordPath).asInt();
+                log.debug("Number of records to harvest: " + numberOfRecordsToHarvest);
+            } catch (Exception e) {
+            }
+        }
+        boolean error = false;
+        HarvestResult result = null;
+        Map<String, Element> allUuids = new HashMap<String, Element>();
+        try {
+            Aligner aligner = new Aligner(cancelMonitor, context, params, log);
+            List<String> listOfUrlForPages = buildListOfUrl(params, numberOfRecordsToHarvest);
+            for (int i = 0; i < listOfUrlForPages.size(); i ++) {
+                if (i != 0) {
+                    jsonResponse = retrieveUrl(listOfUrlForPages.get(i), log);
+                    jsonObj = objectMapper.readTree(jsonResponse);
+                }
+                Map<String, Element> uuids = new HashMap<String, Element>();
+                JsonNode nodes;
+                if (StringUtils.isNotEmpty(params.loopElement)) {
+                    try {
+                        nodes = jsonObj.at(params.loopElement);
+                        log.debug("Number of records in response: " + nodes.size());
+
+                        nodes.forEach(record -> {
+                            Element xml = convertRecordToXml(record);
+                            uuids.put(record.get(params.recordIdPath).asText(), xml);
```

Comment:
@fxprunayre, @josegar74 if the identifier is an URI (like in ESRI DCAT eg `"identifier": "https://data-atmo-hdf.opendata.arcgis.com/datasets/bac17d7d05a34242a8b22c535ecdb13d"` it set the URI as the uuid and it does not work when I open the metadata page.
I did a hack for my test:
```java
uuids.put(record.get(params.recordIdPath).asText().split("/datasets/")[1], xml);
```
but I don't know how we could handle that properly, in any situation, would you have any suggestion ?
having a regexp in the harvester setting to extract the uuid but seems a bit tricky for the admins.
Thanks

Reply:
@fxprunayre did this PR https://github.com/geonetwork/core-geonetwork/pull/5736 that probably helps, but needs to enable some specific configuration.

If the identifiers have this format: http(s)://URL/UUID, maybe an option is when converting the JSON to ISO19139, set the `gmd:fileIdentifier` to the UUID part of the identifier element in JSON and store the full idenfiier in `gmd:identifier` element. That should not require any hack in the UI code, but not sure if it's really "correct" from the metadata content point of view.

Reply:
Thanks @josegar74, yes I think it's the way to go
I would just keep the uuid for the uuid and geep the uri for the resourceIdentifier.
But it means while harvesting, I have to know where is the uuid in the URI to extract it.